src/main/python/pre_fx_data.py

```diff
@@ -0,0 +1,269 @@
+import json
+import logging
+import copy
+from typing import Optional, Dict, Any
+
+from PyQt5.QtCore import QObject, pyqtSignal
+
+from ipfx.ephys_data_set import EphysDataSet
+from ipfx.qc_feature_extractor import cell_qc_features, sweep_qc_features
+from ipfx.qc_feature_evaluator import qc_experiment, DEFAULT_QC_CRITERIA_FILE
+from ipfx.bin.run_qc import qc_summary
+from ipfx.stimulus import StimulusOntology, Stimulus
+from ipfx.data_set_utils import create_data_set
+
+from error_handling import exception_message
+
+
+class PreFxData(QObject):
```

Comment:
Would 'QCData' be a better name?

Reply:
maybe `AutoQcData`? It also includes the actual ephys dataset though ...